src/mergeplg/merge.py

```diff
@@ -114,6 +118,194 @@ def calculate_cml_geometry(ds_cmls, disc=8):
     # Store x and y coordinates in the same array (n_cmls, y/x, disc)
     return np.array([ypos, xpos]).transpose([1, 0, 2])
 
+class Merge:
+    def __init__(self, ds_cml, ds_rad, grid_location_radar, min_obs, smooth_factor):
+        # Check that both time dimensions are similar
+        if not (ds_cml.time == ds_rad.time).all():
+            raise ValueError 
+        
+        # The dataframes are just referenced, not stored twice
+        self.ds_cml = ds_cml 
+        self.ds_rad = ds_rad
+        
+        # Stor hyperparameters
+        self.min_obs, self.smooth_factor = min_obs, smooth_factor
+        
+        # Check that CMLs are inside radar grid?
+        
+        # Calculate intersect weights
+        intersect_weights = plg.spatial.calc_sparse_intersect_weights_for_several_cmls(
+            x1_line=ds_cml.site_0_lon.data,
+            y1_line=ds_cml.site_0_lat.data,
+            x2_line=ds_cml.site_1_lon.data,
+            y2_line=ds_cml.site_1_lat.data,
+            cml_id=ds_cml.cml_id.data,
+            x_grid=ds_rad.longitudes.data,
+            y_grid=ds_rad.latitudes.data,
+            grid_point_location=grid_location_radar,  
+        )
+        
+        # Calculate radar ranfall along CMLs        
+        self.R_rad = plg.spatial.get_grid_time_series_at_intersections(
+            grid_data=ds_rad.rainfall_amount,
+            intersect_weights=intersect_weights,
+        )
+        
+        # Store x and y grids for faster access later
+        self.xgrid, self.ygrid = ds_rad.xs.data, ds_rad.ys.data
+        
+class MergeAdditiveIDW(Merge):
+    def __init__(self, ds_cml, ds_rad, grid_location_radar='center', min_obs=5, smooth_factor=3):
+        Merge.__init__(self, ds_cml, ds_rad, grid_location_radar, min_obs, smooth_factor)
+        
+        # Calculate the difference between radar and CML
+        self.R_diff = xr.where(
+            (self.R_rad > 0) & (self.ds_cml.R > 0), 
+            self.ds_cml.R - self.R_rad, 
+            np.nan,
+        )
+        
+        self.R_diff['mid_x'] = (ds_cml.site_0_x + ds_cml.site_1_x) / 2
+        self.R_diff['mid_y'] = (ds_cml.site_0_y + ds_cml.site_1_y) / 2
+
+    def __call__(self, time, smooth_factor=1):
```

Comment:
looks like the smooth_factor is not needed here, also not in the Kriging class below. Right?